### ISSUE TITLE
Error when creating Table of Contents and there is a duplicate

### ISSUE THREAD

**Issue by MCFHTAGENTS** [EWSoftware/SHFB#972]
I am seeing this error in my build:
````
SHFB: Error BE0029: Unexpected error while executing plug-in &#39;Website Table of Contents Generator&#39;: System.ArgumentException: An item with the same key has already been added.
   at System.ThrowHelper.ThrowArgumentException(ExceptionResource resource)
   at System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
   at System.Linq.Enumerable.ToDictionary[TSource,TKey,TElement](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector, IEqualityComparer`1 comparer)
   at SandcastleBuilder.PlugIns.WebsiteTableOfContentsGeneratorPlugIn.Execute(ExecutionContext context)
   at SandcastleBuilder.Utils.BuildEngine.BuildProcess.ExecutePlugIns(ExecutionBehaviors behavior)
   at SandcastleBuilder.Utils.BuildEngine.BuildProcess.ExecutePlugIns(ExecutionBehaviors behavior)
   at SandcastleBuilder.Utils.BuildEngine.BuildProcess.Build()

An item with the same key has already been added.
   at System.ThrowHelper.ThrowArgumentException(ExceptionResource resource)
   at System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
   at System.Linq.Enumerable.ToDictionary[TSource,TKey,TElement](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector, IEqualityComparer`1 comparer)
   at SandcastleBuilder.PlugIns.WebsiteTableOfContentsGeneratorPlugIn.Execute(ExecutionContext context)
   at SandcastleBuilder.Utils.BuildEngine.BuildProcess.ExecutePlugIns(ExecutionBehaviors behavior)
````
It would be helpful if this could be trapped and you write out the duplicate values in the log so I can determine what is wrong.
Or pick one of them

**Comment by EWSoftware** [comment]
There shouldn't be duplicates in the TOC so the underlying cause should be addressed rather than just hiding it by picking one.  Can you supply an example so that I can duplicate the issue and fix it?

**Comment by MCFHTAGENTS** [comment]
Eric,
Agreed there should not be duplicates but the helpfile is large and so was hoping to get an error message to allow me to locate the duplicate! Unless there's an intermediate file I can share that will help with the root cause.

Have switched off the TOC for now as I don't need it. What's the easiest way to get a sample project to you?




Thank you
Mark

**Comment by EWSoftware** [comment]
You can attach it here or provide a download link to it.  Either can be sent to me via e-mail if you'd prefer.

**Comment by MCFHTAGENTS** [comment]
Hi - can we do email please

**Comment by MCFHTAGENTS** [comment]
[SandcastleDummy.zip](https://github.com/EWSoftware/SHFB/files/10777586/SandcastleDummy.zip)

**Comment by EWSoftware** [comment]
I just ran into the duplicate TOC key issue while looking into the other issue.  The cause was having the framework version in the SHFB project set incorrectly.  You're documenting .NET 6/7 assemblies so it should be set to ".NET Core/.NET Standard/.NET 5.0+".  When set to .NET Framework 4.7.2, it uses the wrong set of assemblies to generate the reflection data.  Sometimes it appears to work and other times it crashes while generating the reflection data.  In this case, it made it past that but the info is incorrect resulting in the duplicates.

If using a solution or project as the documentation source, it can auto-correct the setting at build time as it can see what it's supposed to use in the projects but when using assemblies directly as documentation sources, it can't do that.  Changing the SHFB project's framework version as indicated should resolve this issue.  I've updated the plug-in to report duplicates as a warning with a note to check the framework version in the SHFB project.
